### PR TITLE
Add new Header-Values for Component Analysis api-call

### PR DESCRIPTION
**Pull request by dvandra**
Add new Header-Values for Component Analysis api-call:
  - 'security_vendor' : 'snyk',
  - 'version' : '0.0.14',

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -227,6 +227,8 @@ const get_metadata = (ecosystem, name, version) => {
                 }
                 options['headers'] = {
                     'Authorization' : 'Bearer ' + config.api_token,
+                    'security_vendor' : 'snyk',
+                    'version' : '0.0.14',
```
Comment (arajkumar) [inline src/server.ts]: We shouldn't hardcode any vendor specific fields over here. Better pass these headers from plugin through env variable or other IPC mechanism recommended by vscode. You can look at how we are passing 3scale API endpoint and user key from plugin to lsp, which uses env variable based approach.

File: src/server.ts [hunk 2]
```diff
@@ -206,7 +206,7 @@ const getCAmsg = (deps, diagnostics): string => {
 };
 
 const caDefaultMsg = 'Checking for security vulnerabilities ...';
-
+const custom_header = JSON.parse(process.env.CUSTOM_HEADER);
```
Comment (arajkumar) [inline src/server.ts]: Follow the existing coding standard of the file.

Use camelcase for naming variables.

File: src/server.ts [hunk 3]
```diff
@@ -227,6 +227,8 @@ const get_metadata = (ecosystem, name, version) => {
                 }
                 options['headers'] = {
                     'Authorization' : 'Bearer ' + config.api_token,
+                    'security-vendor' : custom_header.security_vendor,
```
Comment (arajkumar) [inline src/server.ts]: You shouldn't explicitly retrieve the custom headers fields, it defeats the purpose of passing it as JSON.You _should_ concat those.

`options['headers'].concat(customHeader);`

**Comment by dvandra** [comment]
Closing this PR. 
#116

**Pull request closed without merging**